RocketChat/Rocket.Chat.ReactNative
# fix: link preview without embed image

## Proposed changes

The behavior of embedded link previews is not the same between the regular browser client and the official mobile (Android/iPhone) client. What I’m seeing is that when I paste a directly-linked image (ending in .jpg or similar), the browser client will display a preview for it properly. In the mobile client, there will be no preview shown.

The Autolinker function is similarly inconsistent, and changing the Regex field doesn’t seem to have any effect. A good example is direct links for https://i.redd.it/ images. These are hyperlinked and previewed properly on the desktop client, no issues there. However, on the mobile client these links are NOT hyperlinked and no preview is shown. I’ve tried editing the RegEx field in Settings to account for [i.redd.it](http://i.redd.it/) links, nothing I do seems to make a difference. I’ve tried restarting the RocketChat instance after changing it, no difference.

SOME links do get a preview properly in the mobile client, but not direct image links. If I link [Welcome to Dilbert](http://dilbert.com/), for example, that gets a preview in the mobile client just fine. YouTube videos get an embed preview as well.

I'm mainly testing on my Android phone, but I've done a few side tests with an iPhone and run into the same problem. I have also tested with the browser client through Chrome on the Android phone, and it behaves as I would expect.

## Issue(s)	

Closes #4482 

## How to test or reproduce

- Go to a channel and post a direct url to an image. In my example, https://i.redd.it/t2mul61342l91.jpg
- In the browser client, observe that the url is hyperlinked and a link preview is shown.
- In the mobile client, observe that the url is bare, no hyperlink, no link preview shown.

## Screenshots

| Before | After |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/47038980/205175493-fc1f7fdd-d10a-4099-88c4-146bac69e223.png) | ![image](https://user-images.githubusercontent.com/47038980/205175462-ec36913a-1ea7-458c-ae41-94ddf652d933.png) |


## Types of changes




## Checklist



## Further comments

[TC-165]

[TC-165]: https://rocketchat.atlassian.net/browse/TC-165?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
